src/box/lua/config/descriptions.lua

```diff
@@ -0,0 +1,2446 @@
+    of a follower in the current election term and becomes read-only.
+
+    - In `soft` mode, a connection is considered dead if there are no responses
+    for 4 * `replication.timeout` seconds both on the current leader and the
+    followers.
+    - In `strict` mode, a connection is considered dead if there are no
+    responses for 2 * `replication.timeout` seconds on the current leader
+    and 4 * `replication.timeout` seconds on the followers. This improves the
+    chances that there is only one leader at any time.
+
+    Fencing applies to the instances that have the `replication.election_mode`
+    set to `candidate` or `manual`. To turn off leader fencing, set
+    `election_fencing_mode` to off.
+]])
+
+M['replication.election_mode'] = format_text([[
+    A role of a replica set node in the leader election process.
+
+    The possible values are:
+
+    - `off`: a node doesn't participate in the election activities.
+    - `voter`: a node can participate in the election process but can't be
+    a leader.
+    - `candidate`: a node should be able to become a leader.
+    - `manual`: allow to control which instance is the leader explicitly
+    instead of relying on automated leader election. By default, the instance
+    acts like a voter - it is read-only and may vote for other candidate
+    instances. Once `box.ctl.promote()` is called, the instance becomes a
+    candidate and starts a new election round. If the instance wins the
+    elections, it becomes a leader but won't participate in any new elections.
+]])
+
+M['replication.election_timeout'] = format_text([[
+    Specifies the timeout (in seconds) between election rounds in the leader
+    election process if the previous round ended up with a split vote.
+
+    It is quite big, and for most of the cases, it can be lowered to 300-400 ms.
+
+    To avoid the split vote repeat, the timeout is randomized on each node
+    during every new election, from 100% to 110% of the original timeout value.
+    For example, if the timeout is 300 ms and there are 3 nodes started the
+    election simultaneously in the same term, they can set their election
+    timeouts to 300, 310, and 320 respectively, or to 305, 302, and 324, and
+    so on. In that way, the votes will never be split because the election on
+    different nodes won't be restarted simultaneously.
+]])
+
+M['replication.failover'] = format_text([[
+    A failover mode used to take over a master role when the current master
+    instance fails. The following modes are available:
+
+    - `off`: Leadership in a replica set is controlled using the
+    `database.mode` option. In this case, you can set the `database.mode` option
+    to rw on all instances in a replica set to make a master-master
+    configuration.
+
+    The default database.mode is determined as follows: `rw` if there is one
+    instance in a replica set; `ro` if there are several instances.
```

Comment:
It seems, we're going to use some subset of Markdown features. I think it is appropriate to give some Markdown related suggestions here.

First, the list items can be single-paragraph or multi-paragraph. For example:

- A single-paragraph list item.
- A multi-paragraph list item.

  This is the second paragraph in this list item.

Source:

```markdown
- A single-paragraph list item.
- A multi-paragraph list item.

  This is the second paragraph in this list item.
```

So, if we indent a paragraph to the same level as the list item's text, it becomes a part of the list item.

We can use it here to highlight that some paragraphs are about a particular `replication.failover` mode.

Also, a remark about single-paragrph nodes with line breaks. There are two formatting variants:

```markdown
- foo bar baz fiz
lorem ipsum
```

```markdown
- foo bar baz fiz
  lorem ipsum
```

(They renders the same HTML.)

As I see, you're using the first variant. Personally I prefer the second, because in this case the indent visually highlight that the given text corresponds to the given list item.

Feel free to refuse to reformat a lot of the text in this pull request. I've written it down to share thoughts.

Reply:
I have rewritten all the descriptions to the second variant.

Reply:
Nice, thanks!